nvflare/fuel/utils/log_utils.py

```diff
@@ -240,33 +240,34 @@ def format(self, record) -> str:
         super().format(record)
 
         self.record.asctime = self.formatTime(self.record, self.datefmt)
-        formatted_message_dict = self.formatMessage(self.record)
+        formatted_message_dict = self.formatMessageDict(self.record)
         message_dict = {k: v for k, v in formatted_message_dict.items()}
 
         return json.dumps(message_dict, default=str)
 
 
 class LoggerNameFilter(logging.Filter):
-    def __init__(self, logger_names=["nvflare"], exclude_logger_names=[]):
+    def __init__(self, logger_names=["nvflare"], exclude_logger_names=[], allow_errors=True):
         """Filter log records based on logger names.
-        Additionally allows all log records with levelno > logging.INFO through.
 
         Args:
             logger_names (List[str]): list of logger names to allow through filter
             exclude_logger_names (List[str]): list of logger names to disallow through filter (takes precedence over allowing from logger_names)
+            allow_errors (bool): allow all log records with levelno > logging.INFO through. Defaults to True.
 
         """
         super().__init__()
         self.logger_names = logger_names
         self.exclude_logger_names = exclude_logger_names
+        self.allow_errors = allow_errors
 
     def filter(self, record):
         name = getattr(record, "fullName", record.name)
 
         is_logger_included = self.matches_name(name, self.logger_names)
         is_logger_excluded = self.matches_name(name, self.exclude_logger_names)
 
-        return (record.levelno > logging.INFO) or (not is_logger_excluded and is_logger_included)
+        return (self.allow_errors and record.levelno > logging.INFO) or (is_logger_included and not is_logger_excluded)
```

Comment:
just notice this right now,
why would we have both "included" and "excluded"?
Should one be enough?

Reply:
If we want, we can remove "excluded" for more simplicity. The idea was since logger names are hierarchical, you could include "a" but exclude "a.b.c" for example, which gives more fine-grained control. Otherwise with only include this would be much more difficult, but maybe it is not a common use case.